File: Chapter 05/C#/Section_5_4_Server_Monitor/After/Infrastructure/Adapters/KafkaAlertAdapter.cs
using System;
using System.Text.Json; // Native .NET JSON
using Chapter05.ServerMonitor.After.Core.Ports;
using Chapter05.ServerMonitor.After.Infrastructure.ExternalLibs; // Fake 3rd party lib

namespace Chapter05.ServerMonitor.After.Infrastructure.Adapters
{
    // ADAPTER 3: The "Scale" Adapter (Async Messaging)
    public class KafkaAlertAdapter : IAlertPort
    {
        private readonly IProducer<string, string> _kafkaProducer;

        public KafkaAlertAdapter(IProducer<string, string> kafkaProducer)
        {
            _kafkaProducer = kafkaProducer;
        }

        public void SendAlert(string message)
        {
            var payload = new { Error = message, Timestamp = DateTime.UtcNow };
            var json = JsonSerializer.Serialize(payload);
            
            // Fire and forget! The Core doesn't need to wait for an ACK.
            _kafkaProducer.Produce("server-alerts-topic", json);
        }
    }
}

File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/after/infrastructure/adapters/KafkaAlertAdapter.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.infrastructure.adapters;

import com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.core.ports.AlertPort;
import com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.infrastructure.externallibs.Producer;
import java.time.Instant;

/**
 * ADAPTER 3: The "Scale" Adapter (Async Messaging).
 */
public class KafkaAlertAdapter implements AlertPort {
    private final Producer<String> kafkaProducer;

    public KafkaAlertAdapter(Producer<String> kafkaProducer) {
        this.kafkaProducer = kafkaProducer;
    }

    @Override
    public void sendAlert(String message) {
        // In a real app, use a JSON library like Jackson
        String payload = String.format("{\"Error\": \"%s\", \"Timestamp\": \"%s\"}", 
                                        message, Instant.now());
        
        kafkaProducer.produce("server-alerts-topic", payload);
    }
}